src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -36,6 +35,16 @@ class LazyList[+A](expr: => LazyList.Evaluated[A])
 
   protected[this] def fromSpecificIterable(coll: collection.Iterable[A]): LazyList[A] = fromIterable(coll)
 
+  protected[this] def toCollection: LazyList[A] => IterableOnce[A] = identity
+
+  def updated[B >: A](index: Int, elem: B): LazyList[B] =
+    zipWithIndex.map { case (a, i) => if (i == index) elem else a }
+
+  def zipWithIndex: LazyList[(A, Int)] =
```

Comment:
There ought to be a more efficient way to implement this at a lower level but we can leave that for a future performance improvement sprint.